pkg/controller/node/nodecontroller.go

```diff
@@ -598,6 +667,10 @@ func (nc *NodeController) monitorNodeStatus() error {
 			zoneToNodeConditions[utilnode.GetZoneKey(node)] = append(zoneToNodeConditions[utilnode.GetZoneKey(node)], currentReadyCondition)
 		}
 
+		if nc.useTaintBasedEvictions {
+			nc.syncupNodeForNoScheduleTainting(node)
```

Comment:
We shouldn't do this in 'monitorNodeStatus', as it runs every 10s (or sth). This would mean that there can be designed 10s delay between Node getting a Condition and Taint appearance, which is not enough. For those kind of taints we just need to register a function in Informer that will fire on every Node create and update.